packages/bridge/src/interface.ts

```diff
@@ -41,6 +41,24 @@ export interface BridgeProvider {
     params: GetBridgeQuoteParams
   ) => Promise<BridgeTransactionRequest>;
 
+  /**
+   * Requests the available source assets for a given destination chain and destination coin.
+   * Restricted to assets that don't change the user's underlying asset exposure, in other words, is the same variant of the asset.
+   * In practice, this can be used to offer a list of selectable assets for the user to choose from.
+   *
+   * In general should avoid throwing errors, but return an empty array if no source assets are available with the given input.
+   * May throw an error if an unexpected error occurs.
+   *
+   * @param toChain The destination chain.
+   * @param toAsset The destination coin.
+   * @returns A promise that resolves to an array of source coins with source identifiers.
+   * @throws Unexpected errors.
+   */
+  getAvailableSourceAssetVariants(
+    toChain: BridgeChain,
+    toAsset: BridgeAsset
+  ): Promise<(BridgeChain & BridgeAsset)[]>;
```

Comment:
Nit: In line with our other methods, I think we should make the parameters a schema or an object. This will make it easier to understand and manage the parameters

Reply:
IMO it's more succinct to have a set of composable types that are single purpose that can be combined like this when necessary. In this case, it returns a bridge chain and asset together (the combo)

Reply:
In general I think the codebase is simpler when there's overall fewer symbols/functions/defs vs when there's more. If we can avoid a def, then it's easier to grok/navigate

Reply:
Maybe we can inline then these parameters if we want to avoid creating a schema. 

```suggestion
  getAvailableSourceAssetVariants(params: {
    toChain: BridgeChain,
    toAsset: BridgeAsset
  }): Promise<(BridgeChain & BridgeAsset)[]>;
```